### DISCUSSION TITLE
Problems connecting web sync

### DISCUSSION THREAD

**Discussion by ItsSiem** [gramps-project/gramps-web#375] [General]
Whenever I try to connect to my gramps web instance through it's assigned domain I get the insufficient permissions error. However when I use the local IP address to connect it works just fine. This had me thinking my reverse proxy or something like that was not set up correctly. But I can access and use my instance just fine using the domain. I also tried using the API directly and that seems to work fine too.

I'm unsure if this is a bug in gramps-web or in the addon, or if it even is a bug at all. It's also definitely possible that there is just something wrong in my config.

Presuming this is a configuration issue, is there any way that I can get a log from the addon? That would probably help a lot.

**Comment by DavidMStraub** [comment]
Hi, I convert this into a discussion as it is certainly not an issue with the Gramps Web frontend. It could be a configuration issue or a bug in the addon.

But you will need to provide more details. Which versions are you using? How did you install them? What is the permission level of your user account? (You need owner permissions for synchronization.)

**Reply by ItsSiem** [reply]
Hi, I'm using the latest gramps-web and it has been deployed using docker compose. My instance is set up with two trees and I have an account on both trees with owner privileges.

Gramps Web Config:
Gramps 5.1.6
Gramps Web API 1.5.1
Gramps Web Frontend 24.1.2
locale: en
multi-tree: true
task queue: true

Regarding the rest of the chain, I use NGINX as a reverse proxy and I also have a valid SSL certificate installed.

**Reply by DavidMStraub** [reply]
What is the exact error (screenshot)? Please check your container logs while the sync addon is running.

**Reply by ItsSiem** [reply]
![image](https://github.com/gramps-project/gramps-web/assets/47294812/dd5f8fbf-bdaa-4ced-9a4e-2cc0ad4e0464)

The logs do not show anything out of the ordinary.

```
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.

INFO  [alembic.runtime.migration] Will assume non-transactional DDL.

[2024-01-30 22:18:35 +0000] [12] [INFO] Starting gunicorn 21.2.0

[2024-01-30 22:18:35 +0000] [12] [INFO] Listening at: http://0.0.0.0:5000 (12)

[2024-01-30 22:18:35 +0000] [12] [INFO] Using worker: sync

[2024-01-30 22:18:35 +0000] [13] [INFO] Booting worker with pid: 13

[2024-01-30 22:18:35 +0000] [14] [INFO] Booting worker with pid: 14

[2024-01-30 22:18:35 +0000] [15] [INFO] Booting worker with pid: 15

[2024-01-30 22:18:35 +0000] [17] [INFO] Booting worker with pid: 17

[2024-01-30 22:18:35 +0000] [16] [INFO] Booting worker with pid: 16

[2024-01-30 22:18:35 +0000] [18] [INFO] Booting worker with pid: 18

[2024-01-30 22:18:35 +0000] [19] [INFO] Booting worker with pid: 19

[2024-01-30 22:18:36 +0000] [20] [INFO] Booting worker with pid: 20
```

**Comment by jerrod213** [comment]
Was this ever solved? I'm having the same error.  I'm using NGINX on my local PC to serve the files through Cloudflare.  I have Cloudflare tunnels setup with high security, so it might be something related to how the sync add on connects.  I was also able to connect locally, so that really takes care of the issue for me as the admin.  If any of my family want to use the desktop app and sync it, there will be an issue.

**Reply by DavidMStraub** [reply]
Sorry, but that not clear what the question is, and not enough detail to provide any help.

If you are facing issues, please open a new discussion at https://github.com/DavidMStraub/gramps-web-sync/ and provide all the details about your setup.

**Comment by chaseblock** [comment]
If using Cloudflare as a proxy, they may be blocking the API access due to user agent blocking or browser integrity checks. Disabling these checks (and possibly others) under the `Security -> WAF` settings on your Cloudflare dashboard may help.

**Reply by matthewthowells** [reply]
Can confirm this was the fix in my case - creating a WAF rule and disabling just User Agent Blocking and Browser Integrity Checks for the subdomain was enough to get the web sync to work. Thanks a bunch!

Tip for anyone else using Cloudflare as a proxy (which I wish I'd known earlier...) you can also see under `Security -> Events` what Cloudflare is blocking. In my case, it was Browser Integrity Checks.

**Reply by Tohtli** [reply]
That was my case also, using Cloudflare to administer the domain and caching the addresses, and Nginx Proxy Manager as a proxy. The solution was to configure a custom rule with the skip action on Cloudflare to disable the Browser Integrity Check and it solved my connection issues between the plugin and the web server
